url80587fc7d5_dennervidal_CloudSim_DVFS_tgz-pJ8-Examples_Dvfs_example_simpleJ8/src/org/cloudbus/cloudsim/workflow/DvfsPolicy_2.java

```diff
@@ -0,0 +1,379 @@
+package org.cloudbus.cloudsim.workflow;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Hashtable;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map.Entry;
+
+import org.cloudbus.cloudsim.CloudletSchedulerTimeShared;
+import org.cloudbus.cloudsim.Log;
+import org.cloudbus.cloudsim.Vm;
+import sun.nio.cs.ext.ISCII91;
+
+/**
+ * This class implements an "ideal" scheduler that creates 1 VM for each task.
+ * The VM is the best possible. The execution time generated by this schedule
+ * is used for deadline generation.
+ *
+ */
+
+
+public class DvfsPolicy_2 extends Policy {
+			
+	List<Vm> vmOffersList;
+	OneCriticalPath TheCP;
+	int vmId=0;
+ 
+	 ArrayList<Task> optimzedTasks;
+        
+        @Override
+	public void doScheduling(long availableExecTime, VMOffers vmOffers) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.